yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -228,6 +228,7 @@ private[spark] class ApplicationMaster(
   }
 
   private def sparkContextStopped(sc: SparkContext) = {
+    sc.requestTotalExecutors(0, 0, Map.empty)
```

Comment:
right... I am little mix with yarnScheduler and yarnSchedulerBackend class...
the stop order was:
YarnClusterScheduler.stop
 ->TaskSchedulerImpl.stop -> CoarsedGrainSchedulerbackend.stop -> asyn: stopExecutors
 -> ApplicationMaster.sparkContextStopped(sc)
 
may be it is good way to override stop() in YarnSchedulerBackend. 
```
YarnSchedulerBackend.stop {
  doRequestTotalExecutors() //this is a askWithRetry...
  super.stop
}
```